src/Section.php

```diff
@@ -22,15 +22,19 @@ class Section implements Node
     /** @var string[] */
     public array $attributes;
 
-    public function __construct(Node $parent, string $title = '', string $url = '')
+    public function __construct(Node $parent, string $title = '', string $url = '', ?array $attributes = null)
     {
         $this->parent = $parent;
 
         $this->title = $title;
         $this->url = $url;
 
         $this->visible = true;
-        $this->attributes = [];
+        if ($attributes) {
```

Comment:
Could you replace this by a ternary? I like to avoid `else` everywhere.

Reply:
Done. :D